packages/wonder-blocks-date-picker/src/util/temporal-locale-utils.ts

```diff
@@ -0,0 +1,395 @@
+        try {
+            const parsed = parseWithFormat(str, fmt, locale);
+            if (parsed) {
+                return parsed;
+            }
+        } catch {
+            // Try next format
+            continue;
+        }
+    }
+
+    return undefined;
+}
+
+// Helper function to get modifiers for a given day
+export const getModifiersForDay = (
+    day: Date,
+    modifiers: Partial<CustomModifiers>,
+): Array<string> => {
+    const matchedModifiers: Array<string> = [];
+
+    for (const [modifierName, matcher] of Object.entries(modifiers)) {
+        if (!matcher) {
+            continue;
+        }
+
+        // If matcher is a function, call it with the day
+        if (typeof matcher === "function") {
+            if (matcher(day)) {
+                matchedModifiers.push(modifierName);
+            }
+        }
+        // If matcher is a Date, check if it's the same day
+        else if (matcher instanceof Date) {
+            if (
+                day.getFullYear() === matcher.getFullYear() &&
+                day.getMonth() === matcher.getMonth() &&
+                day.getDate() === matcher.getDate()
+            ) {
+                matchedModifiers.push(modifierName);
+            }
+        }
+    }
+
+    return matchedModifiers;
+};
+
+/**
+ * Convert a Temporal.PlainDate to a JavaScript Date object.
+ * Sets the time to midnight in the local timezone.
+ */
+export function temporalDateToJsDate(date: Temporal.PlainDate): Date {
+    return new Date(date.year, date.month - 1, date.day);
+}
+
+/**
+ * Convert a JavaScript Date to a Temporal.PlainDate.
+ * Uses local date components (not UTC).
+ */
+export function jsDateToTemporalDate(date: Date): Temporal.PlainDate {
+    return Temporal.PlainDate.from({
+        year: date.getFullYear(),
+        month: date.getMonth() + 1,
+        day: date.getDate(),
+    });
+}
+
+/**
+ * Parse a date string and return a JavaScript Date.
+ * This is a convenience wrapper around parseDate that converts the result
+ * to a Date object for compatibility with react-day-picker.
+ * If a Date is passed in, it's returned as-is.
+ */
+export function parseDateToJsDate(
+    value: string | Date,
+    format: string | Array<string> | null | undefined,
+    locale?: string | null | undefined,
+): Date | null | undefined {
+    // If already a Date, return it
+    if (value instanceof Date) {
+        return value;
+    }
+
+    const temporalDate = parseDate(value, format, locale || undefined);
+    return temporalDate ? temporalDateToJsDate(temporalDate) : undefined;
+}
+
+/**
+ * Get the first day of the week for a given locale.
+ * Returns 0 for Sunday, 1 for Monday, etc.
+ */
+export function getFirstDayOfWeek(locale?: string): number {
+    // Most locales use Monday (1), but some like US use Sunday (0)
+    // If no locale is provided, default to Monday
+    if (!locale) {
+        return 1; // Monday (most of the world)
+    }
+
+    // US, Canada, and some others use Sunday
```

Comment:
TIL Most places start their week on Mondays!

Reply:
I was looking into see if there is a built-in browser API we could use so we don't have to maintain this logic! I learned about [Intl.Locale(locale).weekInfo](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getWeekInfo#firstday)! It doesn't have full browser support too, I'm not sure if it is covered by temporal polyfill!

Reply:
Good question! I added this to the Locale ticket too: https://khanacademy.atlassian.net/browse/FEI-7342

Reply:
We don't actually need this, as that function was added and unused! I've removed it from the PR.